posts/ultra-portable-arm64/index.md

```diff
@@ -0,0 +1,185 @@
+
+## Solution: An anywhere ARM64 dev environment
+
+What I’d like is an ultra-portable ARM64 device that I can use for development. By “portable” I mean that’s it’s available anywhere. The “ultra” part means that I don’t want to have to carry the physical device and its accessories with me. And I’d like the full desktop experience because my project is all about interactivity.
+
+I chose to solve this with Coder and [AWS EC2 Graviton](https://aws.amazon.com/ec2/graviton/).
+
+- With Coder, I can create a workspace that spins up a container for ARM64 Ubuntu letting me work on my project wherever I want.
+- Graviton is the AWS ARM64-based EC2 service. I’ll run Ubuntu with a desktop environment and access it remotely with VNC.
+- Bonus points for remotely accessing my ultra-portable ARM64 device with just a browser-based VNC client.
+
+This will be an excellent solution while I wait to finally get my hands on a real Raspberry Pi.
+
+## Reuse a wheel instead of reinventing it
+
+A Coder workspace provides a development environment. A Coder template describes what’s in the workspace, how to spin it up, and how to take it down.
+
+Coder comes with a growing list of example templates and there’s an expanding ecosystem of third party templates. Unfortunately I couldn’t find a template that provisions ARM64, Linux, a desktop environment, and VNC.
+
+But I can adapt an existing template pretty easily. To set up my ARM64 desktop workspace, I’ll reuse Coder’s template for [Linux on AWS EC2](https://github.com/coder/coder/tree/main/examples/templates/aws-linux).
+
+This template creates a workspace for an EC2 instance of Ubuntu on amd64. Because a Coder template is just a Terraform file, I can edit the template to meet my needs.
+
+I did this in these steps:
+
+0. Start with a starter template.
+1. Set up AWS credentials.
+2. Create the template.
+3. Use ARM64 instead of amd64.
+4. Install a desktop environment and VNC.
+5. Develop on an ultra-portable ARM64!
+
+I'm doing this incrementally to make things easier. Coder helps me along the way by letting me edit templates in-browser and automatically versioning them. If I make a mistake as I figure this out, I can always roll back.
+
+## 0. Start with a starter template
+
+To start, from my browser I log into my Coder instance running on my homelab, Marvin.  I select **Templates**, **Starter templates**, then **Develop in Linux on AWS EC2**.
+
+![Starting with a starter template](./static/aws-ec2-template-new.png)
+
+A new page shows details about setting up the new template. This includes a sample AWS access policy that lets Coder authenticate its access to AWS EC2.
+
+![Help on creating the template](./static/aws-ec2-template-info.png)
+
+
+## 1. Set up AWS credentials
+
+To start and stop my workspace, Coder needs my AWS credentials. The sample AWS access policy specifies these permissions for me.
```

Comment:
You can link to this doc, which explains why this is necessary: https://coder.com/docs/v2/latest/templates/authentication